modules/scan/class-admin-bar-notice.php

```diff
@@ -0,0 +1,202 @@
+
+	/**
+	 * Get the singleton instance of the class.
+	 *
+	 * @return Admin_Bar_Notice
+	 */
+	public static function instance() {
+		if ( ! isset( self::$instance ) ) {
+			self::$instance = new Admin_Bar_Notice();
+			self::$instance->init_hooks();
+		}
+
+		return self::$instance;
+	}
+	/**
+	 * Initalize the hooks as needed.
+	 */
+	private function init_hooks() {
+		if ( ! $this->should_try_to_display_notice() ) {
+			return;
+		}
+
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_toolbar_script' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_toolbar_script' ) );
+		add_action( 'admin_bar_menu', array( $this, 'add_threats_to_toolbar' ), 999 );
+	}
+
+	/**
+	 * Whether to even try to display the notice or now.
+	 *
+	 * @return bool
+	 */
+	private function should_try_to_display_notice() {
+		// Jetpack Scan is currently not supported on multisite.
+		if ( is_multisite() ) {
+			return false;
+		}
+
+		// Check if VaultPress is active, the assumtion there is that VaultPress is working.
+		// It has its own notice in the admin bar.
+		if ( class_exists( 'VaultPress' ) ) {
+			return false;
+		}
+
+		// Only show the notice to admins.
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Add the inline styles and scripts if they are needed.
+	 */
+	public function enqueue_toolbar_script() {
+		$this->add_inline_styles();
+
+		if ( ! is_null( $this->has_threats() ) ) {
+			return;
+		}
+
+		// We don't know about threats in the cache lets load the JS that fetches the info and updates the admin bar.
+		Assets::enqueue_async_script( self::SCRIPT_NAME, '_inc/build/scan/admin-bar-notice.min.js', 'modules/scan/admin-bar-notice.js', array(), self::SCRIPT_VERSION, true );
+
+		$script_data = array(
+			'nonce'              => wp_create_nonce( 'wp_rest' ),
+			'scan_endpoint'      => get_rest_url( null, 'jetpack/v4/scan' ),
+			'scan_dashboard_url' => Redirect::get_url( 'calypso-scanner' ),
+			/* translators: %s is the alert icon */
+			'singular'           => sprintf( esc_html__( '%s Threat found', 'jetpack' ), $this->get_icon() ),
+			/* translators: %s is the alert icon */
+			'multiple'           => sprintf( esc_html__( '%s Threats found', 'jetpack' ), $this->get_icon() ),
+		);
+		wp_localize_script( self::SCRIPT_NAME, 'Jetpack_Scan', $script_data );
+	}
+
+	/**
+	 * Adds the inline styles if they are needed.
+	 */
+	public function add_inline_styles() {
+		// We know there are no threats so lets not include any css.
+		if ( false === $this->has_threats() ) {
+			return;
+		}
+
+		// We might be showing the threats in the admin bar lets make sure that they look great!
+		$hide_wording_on_mobile = '#wp-admin-bar-jetpack-scan-notice .is-hidden { display:none; } @media screen and (max-width: 959px ) { #wpadminbar #wp-admin-bar-jetpack-scan-notice { width:32px; } #wpadminbar #wp-admin-bar-jetpack-scan-notice a { color: transparent!important; }';
```

Comment:
@ChaosExAnima There's a CSS syntax error here. Opened a PR to fix here in #16155.